packages/apidom-ns-workflows-1/src/predicates.ts

```diff
@@ -60,3 +61,14 @@ export const isSourceDescriptionsElement = createPredicate(
         hasClass('sourceDescriptions', element));
   },
 );
+
+export const isCriterionElement = createPredicate(
+  ({ hasBasicElementProps, isElementType, primitiveEq, hasClass }) => {
+    return (element: unknown): element is CriterionElement =>
+      element instanceof CriterionElement ||
+      (hasBasicElementProps(element) &&
+        isElementType('criterion', element) &&
+        primitiveEq('object', element) &&
+        hasClass('criterion', element));
```

Comment:
There is no class: `criterion` attached to `CriterionElement`.

Reply:
added class